pkg/fleet/installer/packages/datadog_agent_ddot_linux.go

```diff
@@ -140,6 +154,87 @@ func preRemoveDatadogAgentDdot(ctx HookContext) error {
 		if err != nil {
 			log.Warnf("failed to remove DDOT config files: %s", err)
 		}
+
+		// Disable otelcollector in datadog.yaml
+		if err = disableOtelCollectorConfig(); err != nil {
+			log.Warnf("failed to disable otelcollector in datadog.yaml: %s", err)
+		}
+	}
+
+	return nil
+}
+
+// enableOtelCollectorConfig adds otelcollector.enabled: true to datadog.yaml
+func enableOtelCollectorConfig() error {
+	// Read existing config
+	var existingConfig map[string]interface{}
+	data, err := os.ReadFile(datadogYamlPath)
+	if err != nil && !os.IsNotExist(err) {
```

Comment:
Is there a reason why `!os.IsNotExist` is necessary here? I was under the impression that we should always have a `datadog.yaml` present when installing.

Reply:
Yes I think removing it and simplifying the block below would be best:
```
	if err := yaml.Unmarshal(data, &existingConfig); err != nil {
		return fmt.Errorf("failed to parse existing datadog.yaml: %w", err)
	}
```

The datadog-agent package also fails the process if the config file is not present: https://github.com/DataDog/datadog-agent/blob/46ef69f5e294c47a50bbf6d4d16394afd3af5765/pkg/fleet/installer/packages/datadog_agent_linux.go#L460